lib/smack/SmackModuleGenerator.cpp

```diff
@@ -67,8 +67,13 @@ void SmackModuleGenerator::generateProgram(llvm::Module& m) {
       naming.leave();
 
       // First execute static initializers, in the main procedure.
-      if (naming.get(*func) == "main" && rep.hasStaticInits())
-        proc->insert(Stmt::call(SmackRep::STATIC_INIT));
+      if (naming.get(*func) == "main") {
+        proc->insert(Stmt::call(SmackRep::INIT_FUNCS));
+        if(rep.hasStaticInits())
+          proc->insert(Stmt::call(SmackRep::STATIC_INIT));
+      } else if (naming.get(*func).substr(0, 18)  == "__SMACK_init_func_")
+        // TODO? Check for no arguments?
```

Comment:
I think you should add an assertion here that checks that init functions take no arguments and return no values. We should probably enforce this since I cannot imagine that arguments and return values would be needed.